expression/builtin_math_vec.go

```diff
@@ -804,11 +804,38 @@ func (b *builtinAbsUIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinCeilDecToIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCeilDecToIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDecimal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDecimal(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+
+	d := buf.Decimals()
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		res, err := d[i].ToInt()
+		if err == types.ErrTruncated {
```

Comment:
When the ```err == types.ErrOverFlow```, it should be returned err. See [here](https://github.com/pingcap/tidb/blob/master/expression/builtin_math.go#L571) for more information.